core/src/main/java/org/apache/druid/math/expr/Expr.java

```diff
@@ -330,10 +623,10 @@ public String toString()
 abstract class BinaryOpExprBase implements Expr
 {
   protected final String op;
-  protected final Expr left;
-  protected final Expr right;
+  protected Expr left;
+  protected Expr right;
```

Comment:
not sure why final is removed ?

Reply:
Accidental remnants of misguided experiment, will fix, all `Expr` should be immutable